src/azul/indexer.py

```diff
@@ -54,14 +54,27 @@ def settings(self, index_name) -> JSON:
             }
         }
 
+    @classmethod
     @abstractmethod
-    def transformers(self) -> Iterable[Transformer]:
+    def transformers(cls) -> Iterable[Transformer]:
         raise NotImplementedError()
 
     @abstractmethod
     def entities(self) -> Iterable[str]:
         raise NotImplementedError()
 
+    @classmethod
+    def field_types(cls) -> Mapping[str, type]:
+        """
+        Returns a mapping of fields to field types
+
+        :return: dict with nested keys matching Elasticsearch fields and values with the field's type
+        """
+        field_types_ = {}
```

Comment:
Do you need the underscore? I would only use underscore to avoid a conflict but I don't see where the conflict would be.